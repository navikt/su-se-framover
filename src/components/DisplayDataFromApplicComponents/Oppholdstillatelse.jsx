import { Undertittel } from 'nav-frontend-typografi';
import { jaNeiSpørsmål, reverseString } from './HelperFunctions';
import React from 'react';

const Oppholdstillatelse = ({ state }) => {
    return (
        <div style={headerSpacing}>
            <Undertittel style={elementSpacing}>Opplysninger om oppholdstillatelse</Undertittel>

            <div style={sectionGridLayout}>
                <span>Har søker varig oppholdstillatelse:</span>
                <span>{jaNeiSpørsmål(state.oppholdstillatelse.harVarigOpphold)}</span>
            </div>

            <div style={sectionGridLayout}>
                {!state.oppholdstillatelse.harVarigOpphold && (
                    <>
                        <span>Utløpsdato: </span>
                        <span>{reverseString(state.oppholdstillatelse.utløpsDato)}</span>
                    </>
                )}
            </div>

            <div style={sectionGridLayout}>
                {!state.oppholdstillatelse.harVarigOpphold && (
                    <>
                        <span>Har søker søkt om forlengelse?</span>
                        <span>{jaNeiSpørsmål(state.oppholdstillatelse.søktOmForlengelse)}</span>
                    </>
                )}
            </div>
        </div>
    );
};

const sectionGridLayout = {
    marginBottom: '1em',
    display: 'grid',
    gridTemplateColumns: '1fr 1fr'
};

const headerSpacing = {
    marginBottom: '2em'
};

const elementSpacing = {
    marginBottom: '1em'
};

export default Oppholdstillatelse;
