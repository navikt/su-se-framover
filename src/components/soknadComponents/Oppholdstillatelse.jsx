import React, {useState} from "react";
import {Systemtittel} from "nav-frontend-typografi";
import {RadioGruppe, Radio, Feiloppsummering} from 'nav-frontend-skjema';
import {InputFields} from "../FormElements";
import {Hovedknapp} from "nav-frontend-knapper";

const Oppholdstillatelse = ({state, updateField, onClick}) => {

    const [feilmeldinger, setFeilmeldinger] = useState([])

    const updateFunction = name => value => updateField(name, value)

    function midlertidigOppholdstillatelse(){
        if(state.varigopphold === "false"){

           return (
               <div style={{display: 'flex'}}>
                   <InputFields labelText="Oppholdstillatelsens utløpsdato"
                                id={fields.oppholdstillatelseUtløpsdato.htmlId}
                                bredde={"S"}
                                value={state.oppholdstillatelseUtløpsdato || ''}
                                onChange={updateFunction("oppholdstillatelseUtløpsdato")}
                   />
                   <RadioGruppe legend="Har du søkt om forlengelse?">
                       <Radio name="soektforlengelse"
                              label="Ja"
                              value="true"
                              checked={state.soektforlengelse === "true"}
                              onChange={(e => updateField("soektforlengelse", e.target.value))}
                       />

                       <Radio name="soektforlengelse"
                              label="Nei"
                              value="false"
                              checked={state.soektforlengelse === "false"}
                              onChange={(e => updateField("soektforlengelse", e.target.value))}
                       />
                   </RadioGruppe>
               </div>
           )
        }
    }

    return (
        <div>
            <Systemtittel>Opplysninger om oppholdstillatelse</Systemtittel>

            <div>
                <RadioGruppe legend="Har du varig oppholdstillatelse?">
                    <Radio name="varigopphold"
                           label="Ja"
                           value="true"
                           checked={state.varigopphold === "true"}
                           onChange={(e => updateField("varigopphold", e.target.value))}
                    />
                    <Radio name="varigopphold"
                           label="Nei"
                           value="false"
                           checked={state.varigopphold === "false"}
                           onChange={(e => updateField("varigopphold", e.target.value))}
                    />
                </RadioGruppe>
                <div>
                    {
                        midlertidigOppholdstillatelse()
                    }
                </div>
            </div>
            {
                feilmeldinger.length > 0 && <Feiloppsummering tittel={"Vennligst fyll ut mangler"} feil={feilmeldinger} />
            }
            <Hovedknapp onClick={validateForm}>Neste</Hovedknapp>
        </div>
    )

    //------------Lett Validering-----------------------
    function validateForm(){
        const formValues = state
        const errors = validateFormValues(formValues)
        console.log(errors)
        setFeilmeldinger(errors)
        if(errors.length === 0){
            onClick()
        }
    }
}

const fields = {varigopphold: {label: 'varigopphold', htmlId: 'varigopphold'},
    oppholdstillatelseUtløpsdato: {label: 'oppholdstillatelseUtløpsdato', htmlId: 'oppholdstillatelseUtløpsdato'},
    soektforlengelse: {label: 'soektforlengelse', htmlId: 'soektforlengelse'}
}

function validateFormValues(formValues){
    const tempErrors = []

    tempErrors.push(...varigOppholdstillatelseValidering(formValues))
    tempErrors.push(...oppholdstillatelseUtløpsdatoValidering(formValues))
    tempErrors.push(...søktforlengelseValidering(formValues))

    return tempErrors
}

function varigOppholdstillatelseValidering(formValues){
    const varigOppholdstillatelse = formValues.varigopphold
    let feilmelding = ""

    if(varigOppholdstillatelse === undefined){
        feilmelding += "Vennligst velg varig-oppholdstillatelse"
    }
    if(feilmelding.length > 0){
        return [{skjemaelementId: fields.varigopphold.htmlId, feilmelding}]
    }
    return []
}

function oppholdstillatelseUtløpsdatoValidering(formValues){
    const oppholdstillatelseUtløpsdato = formValues.oppholdstillatelseUtløpsdato
    let feilmelding = ""

    if(formValues.varigopphold === "false"){
        if(!/^\d{2}\/\d{2}\/\d{2}$/.test(oppholdstillatelseUtløpsdato)) {
            if(oppholdstillatelseUtløpsdato === '' || oppholdstillatelseUtløpsdato === undefined){
                feilmelding += "Oppholdstillatelsens utløpsdato kan ikke være tom. Den må være i formen dd/mm/yy"
            }else{
                feilmelding += "Oppholdstillatelsens utløpsdato må være i formen dd/mm/yy"
            }
            if(feilmelding.length > 0){
                return [{skjemaelementId: fields.oppholdstillatelseUtløpsdato.htmlId, feilmelding}]
            }
        }
    }
    return []
}

function søktforlengelseValidering(formValues){
    const soektforlengelse = formValues.soektforlengelse
    let feilmelding = ""

    if(formValues.varigopphold === "false"){

        if(soektforlengelse === undefined){
            feilmelding += "Vennligst velg om søker har søkt om forlengelse"
        }
        if(feilmelding.length > 0){
            return [{skjemaelementId: fields.soektforlengelse.htmlId, feilmelding}]
        }
    }
    return []
}

export const validateOppholdstillatelse = {
    validateFormValues
}

export default Oppholdstillatelse;