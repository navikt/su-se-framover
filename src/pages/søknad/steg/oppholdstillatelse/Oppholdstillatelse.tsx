import { yupResolver } from '@hookform/resolvers/yup';
import { Alert, Radio, RadioGroup, TextField } from '@navikt/ds-react';
import { useRef } from 'react';
import { Controller, useForm } from 'react-hook-form';
import { useNavigate } from 'react-router-dom';

import { BooleanRadioGroup } from '~src/components/formElements/FormElements';
import Feiloppsummering from '~src/components/oppsummering/feiloppsummering/Feiloppsummering';
import søknadSlice from '~src/features/søknad/søknad.slice';
import SøknadSpørsmålsgruppe from '~src/features/søknad/søknadSpørsmålsgruppe/SøknadSpørsmålsgruppe';
import { TypeOppholdstillatelse } from '~src/features/søknad/types';
import { focusAfterTimeout } from '~src/lib/formUtils';
import { useI18n } from '~src/lib/i18n';
import { hookFormErrorsTilFeiloppsummering } from '~src/lib/validering';
import { FormData, schema } from '~src/pages/søknad/steg/oppholdstillatelse/validering';
import { useAppDispatch, useAppSelector } from '~src/redux/Store';

import Bunnknapper from '../../bunnknapper/Bunnknapper';
import sharedStyles from '../../steg-shared.module.less';
import sharedI18n from '../steg-shared-i18n';

import messages from './oppholdstillatelse-nb';

const Oppholdstillatelse = (props: { nesteUrl: string; forrigeUrl: string; avbrytUrl: string }) => {
    const harVedtakFraStore = useAppSelector((s) => s.soknad.oppholdstillatelse);
    const dispatch = useAppDispatch();
    const navigate = useNavigate();

    const form = useForm<FormData>({
        resolver: yupResolver(schema),
        defaultValues: harVedtakFraStore,
    });

    const { formatMessage } = useI18n({ messages: { ...sharedI18n, ...messages } });
    const setFieldsToNull = (keys: Array<keyof FormData>) => keys.map((key) => form.setValue(key, null));
    const feiloppsummeringref = useRef<HTMLDivElement>(null);

    return (
        <form
            onSubmit={form.handleSubmit((values) => {
                dispatch(søknadSlice.actions.oppholdstillatelseUpdated(values));
                navigate(props.nesteUrl);
                focusAfterTimeout(feiloppsummeringref)();
            })}
            className={sharedStyles.container}
        >
            <SøknadSpørsmålsgruppe withoutLegend>
                <Controller
                    control={form.control}
                    name="familieforening"
                    render={({ field, fieldState }) => (
                        <BooleanRadioGroup
                            {...field}
                            legend={formatMessage('familieforening.label')}
                            error={fieldState.error?.message}
                            value={field.value}
                        />
                    )}
                />
                {form.watch('familieforening') && (
                    <Alert variant="warning" className={sharedStyles.marginBottom}>
                        {formatMessage('familieforening.info')}
                    </Alert>
                )}

                <Controller
                    control={form.control}
                    name="erNorskStatsborger"
                    render={({ field, fieldState }) => (
                        <BooleanRadioGroup
                            {...field}
                            description={formatMessage('statsborger.description')}
                            legend={formatMessage('statsborger.label')}
                            error={fieldState.error?.message}
                            value={field.value}
                            onChange={(val) => {
                                field.onChange(val);
                                setFieldsToNull(['eøsborger', 'harOppholdstillatelse', 'typeOppholdstillatelse']);
                            }}
                        />
                    )}
                />
                {form.watch('erNorskStatsborger') === false && (
                    <Controller
                        control={form.control}
                        name="eøsborger"
                        render={({ field, fieldState }) => (
                            <BooleanRadioGroup
                                {...field}
                                legend={formatMessage('eøsborger.label')}
                                error={fieldState.error?.message}
                                value={field.value}
                            />
                        )}
                    />
                )}
                {form.watch('eøsborger') && (
                    <Alert variant="warning" className={sharedStyles.marginBottom}>
                        {formatMessage('eøsborger.info')}
                    </Alert>
                )}
                {form.watch('erNorskStatsborger') === false && (
                    <Controller
                        control={form.control}
                        name="harOppholdstillatelse"
                        render={({ field, fieldState }) => (
                            <BooleanRadioGroup
                                {...field}
                                legend={formatMessage('oppholdstillatelse.label')}
                                error={fieldState.error?.message}
                                value={field.value}
                                onChange={(val) => {
                                    field.onChange(val);
                                    setFieldsToNull(['familieforening', 'typeOppholdstillatelse']);
                                }}
                            />
                        )}
                    />
                )}
                {form.watch('harOppholdstillatelse') === false && (
                    <Alert variant="warning" className={sharedStyles.marginBottom}>
                        {formatMessage('oppholdstillatelse.info')}
                    </Alert>
                )}

                {form.watch('harOppholdstillatelse') && (
                    <>
                        <Controller
                            control={form.control}
                            name="typeOppholdstillatelse"
                            render={({ field, fieldState }) => (
                                <RadioGroup
                                    {...field}
                                    legend={formatMessage('typeOppholdstillatelse.label')}
                                    error={fieldState.error?.message}
                                    value={field.value}
                                >
                                    <Radio value={TypeOppholdstillatelse.Midlertidig}>
                                        {formatMessage(TypeOppholdstillatelse.Midlertidig)}
                                    </Radio>
                                    <Radio value={TypeOppholdstillatelse.Permanent}>
                                        {formatMessage(TypeOppholdstillatelse.Permanent)}
                                    </Radio>
                                </RadioGroup>
                            )}
                        />

                        {form.watch('typeOppholdstillatelse') === TypeOppholdstillatelse.Midlertidig && (
                            <Alert variant="warning" className={sharedStyles.marginBottom}>
                                {formatMessage('typeOppholdstillatelse.info')}
                            </Alert>
                        )}
                    </>
                )}

                <Controller
                    control={form.control}
                    name="statsborgerskapAndreLand"
                    render={({ field, fieldState }) => (
                        <BooleanRadioGroup
                            legend={formatMessage('statsborgerskapAndreLand.label')}
                            error={fieldState.error?.message}
                            {...field}
                            onChange={(val) => {
                                field.onChange(val);
                                setFieldsToNull(['statsborgerskapAndreLandFritekst']);
                            }}
                        />
                    )}
                />
                {form.watch('statsborgerskapAndreLand') && (
                    <Controller
                        control={form.control}
                        name="statsborgerskapAndreLandFritekst"
                        render={({ field, fieldState }) => (
                            <TextField
                                className={sharedStyles.narrow}
                                label={formatMessage('statsborgerskapAndreLandFritekst.label')}
                                error={fieldState.error?.message}
                                {...field}
                                value={field.value || ''}
                                autoComplete="off"
                                // Dette elementet vises ikke ved sidelast
                                // eslint-disable-next-line jsx-a11y/no-autofocus
                                autoFocus
                            />
                        )}
                    />
                )}
            </SøknadSpørsmålsgruppe>
            <div>
                <Feiloppsummering
                    className={sharedStyles.marginBottom}
                    tittel={formatMessage('feiloppsummering.title')}
                    feil={hookFormErrorsTilFeiloppsummering(form.formState.errors)}
                    hidden={hookFormErrorsTilFeiloppsummering(form.formState.errors).length === 0}
                    ref={feiloppsummeringref}
                />
            </div>

            <Bunnknapper
                previous={{
                    onClick: () => {
                        dispatch(søknadSlice.actions.oppholdstillatelseUpdated(form.getValues()));
                        navigate(props.forrigeUrl);
                    },
                }}
                avbryt={{
                    toRoute: props.avbrytUrl,
                }}
            />
        </form>
    );
};

export default Oppholdstillatelse;
