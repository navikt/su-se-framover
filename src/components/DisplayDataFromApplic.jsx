import { Systemtittel } from 'nav-frontend-typografi';
import React from 'react';
import Personopplysninger from './DisplayDataFromApplicComponents/Personopplysninger';
import Boforhold from './DisplayDataFromApplicComponents/Boforhold';
import Utenlandsopphold from './DisplayDataFromApplicComponents/Utenlandsopphold';
import Oppholdstillatelse from './DisplayDataFromApplicComponents/Oppholdstillatelse';
import InntektPensjonFormue from './DisplayDataFromApplicComponents/InntektPensjonFormue';
import ForNAV from './DisplayDataFromApplicComponents/ForNAV';

const DisplayDataFromApplic = ({ state }) => {
    console.log(state);
	if(Object.keys(state).length == 0) return null;
    return (
        <div>
            <Systemtittel style={{ marginBottom: '1em' }}>Oppsumerings side</Systemtittel>
            <Personopplysninger state={state} />
            <Boforhold state={state} />
            <Utenlandsopphold state={state} />
            <Oppholdstillatelse state={state} />
            <InntektPensjonFormue state={state} />
            <ForNAV state={state} />
        </div>
    );
};

export default DisplayDataFromApplic;
