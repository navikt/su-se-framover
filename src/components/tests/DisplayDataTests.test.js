import React from 'react';
import { create } from 'react-test-renderer';
import DisplayDataFromApplic from '../DisplayDataFromApplic.jsx';
import Oppholdstillatelse from '../../søknad/steg/Oppholdstillatelse';

const correctState = {
    forNAV: {
        målform: 'Bokmål',
        erPassSjekket: true,
        søkerMøttPersonlig: true,
        harFullmektigMøtt: false,
        forNAVmerknader: 'Alt er fint'
    },
    boforhold: {
        borSammenMed: ['esp', 'over18'],
        delerBolig: true,
        delerBoligMed: [
            { navn: 'Pluto', fødselsnummer: '12345678901' },
            { navn: 'Jupiter', fødselsnummer: '10987654321' }
        ]
    },
    utenlandsopphold: {
        utenlandsopphold: true,
        registrertePerioder: [{ utreisedato: '2020-02-01', innreisedato: '2020-02-02' }],
        planlagtUtenlandsopphold: true,
        planlagtePerioder: [{ utreisedato: '2020-02-04', innreisedato: '2020-02-06' }]
    },
    oppholdstillatelse: { harVarigOpphold: false, søktOmForlengelse: true, utløpsDato: '2020-02-21' },
    personopplysninger: {
        fnr: '12345678901',
        fornavn: 'Planet',
        poststed: 'Melk',
        bokommune: 'Veien',
        etternavn: 'Planetetus',
        flyktning: true,
        bruksenhet: 'H105',
        mellomnavn: 'Planetus',
        postnummer: '0985',
        borFastINorge: true,
        gateadresse: 'Melkeveien 5',
        telefonnummer: '12345678',
        statsborgerskap: 'Solen, Uranus'
    },
    inntektPensjonFormue: {
        sumInntektOgPensjon: 750,
        formueBeløp: '100',
        harPensjon: true,
        harSosialStønad: false,
        framsattKravAnnenYtelse: true,
        pensjonsOrdning: [{ beløp: '250', ordning: 'PensjonsOrdningen' }],
        harFinansFormue: false,
        annenFormueEiendom: [{ typeFormue: 'Hytte', skattetakst: '50' }],
        harFormueEiendom: true,
        harAnnenFormue: true,
        harInntekt: true,
        harDepositumskonto: false,
        framsattKravAnnenYtelseBegrunnelse: 'penjon',
        inntektBeløp: '500'
    }
};

const stateWithError = {
    forNAV: {
        målform: 'Nynorsk',
        erPassSjekket: true,
        søkerMøttPersonlig: true,
        harFullmektigMøtt: false,
        forNAVmerknader: 'Alt er fint'
    },
    boforhold: {
        borSammenMed: ['esp', 'over18'],
        delerBolig: true,
        delerBoligMed: [
            { navn: 'Pluto', fødselsnummer: '12345678901' },
            { navn: 'Jupiter', fødselsnummer: '10987654321' }
        ]
    },
    utenlandsopphold: {
        utenlandsopphold: true,
        registrertePerioder: [{ utreisedato: '2020-02-01', innreisedato: '2020-02-02' }],
        planlagtUtenlandsopphold: true,
        planlagtePerioder: [{ utreisedato: '2020-02-04', innreisedato: '2020-02-06' }]
    },
    oppholdstillatelse: { harVarigOpphold: false, søktOmForlengelse: true, utløpsDato: '2020-02-21' },
    personopplysninger: {
        fnr: '12345678901',
        fornavn: 'Planet',
        poststed: 'Melk',
        bokommune: 'Veien',
        etternavn: 'Planetetus',
        flyktning: true,
        bruksenhet: 'H105',
        mellomnavn: 'Planetus',
        postnummer: '0985',
        borFastINorge: true,
        gateadresse: 'Melkeveien 5',
        telefonnummer: '12345678',
        statsborgerskap: 'Solen, Uranus'
    },
    inntektPensjonFormue: {
        sumInntektOgPensjon: 750,
        formueBeløp: '100',
        harPensjon: true,
        harSosialStønad: false,
        framsattKravAnnenYtelse: true,
        pensjonsOrdning: [{ beløp: '250', ordning: 'PensjonsOrdningen' }],
        harFinansFormue: false,
        annenFormueEiendom: [{ typeFormue: 'Hytte', skattetakst: '50' }],
        harFormueEiendom: true,
        harAnnenFormue: true,
        harInntekt: true,
        harDepositumskonto: false,
        framsattKravAnnenYtelseBegrunnelse: 'penjon',
        inntektBeløp: '500'
    }
};

describe('Tests related to displaying data from a filled application', () => {
    test('Tests that the incoming state has the correct properties', () => {
        const component = create(<DisplayDataFromApplic state={correctState} />);

        expect(component.toTree().props.state).toHaveProperty(
            'forNAV' &&
                'boforhold' &&
                'inntektPensjonFormue' &&
                'personopplysninger' &&
                'utenlandsopphold' &&
                'oppholdstillatelse'
        );
    });

    test('Tests that the component has exactly 7 children', () => {
        const component = create(<DisplayDataFromApplic state={correctState} />);

        //the component should have 7 children.
        expect(component.toJSON().children.length).toBe(7);
    });

    test('Tests that two DisplayDataFromApplic components with a difference in målform are not equal', () => {
        const component = create(<DisplayDataFromApplic state={correctState} />);
        const component2 = create(<DisplayDataFromApplic state={stateWithError} />);

        expect(component.toJSON()).not.toEqual(component2.toJSON());
    });

    test('tests that we use primitive booleans', () => {
        const component = create(<Oppholdstillatelse state={correctState} />);

        expect(component.toTree().props.state.oppholdstillatelse.harVarigOpphold).toBe(false);
    });
});
