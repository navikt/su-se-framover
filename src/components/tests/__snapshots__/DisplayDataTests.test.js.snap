// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests related to displaying data from a filled application Should match its snapshot 1`] = `
<div>
  <h2
    className="typo-systemtittel"
    style={
      Object {
        "marginBottom": "1em",
      }
    }
  >
    Oppsumerings side
  </h2>
  <div
    style={
      Object {
        "marginBottom": "2em",
      }
    }
  >
    <h2
      className="typo-undertittel"
      style={
        Object {
          "marginBottom": "1em",
        }
      }
    >
      Personopplysninger
    </h2>
    <div
      style={
        Object {
          "display": "grid",
          "gridTemplateColumns": "1fr 1fr",
          "marginBottom": "1em",
        }
      }
    >
      <span
        style={
          Object {
            "marginBottom": "1em",
          }
        }
      >
        Fødselsnummer:
      </span>
      <span>
        12345678901
      </span>
    </div>
    <div
      style={
        Object {
          "display": "grid",
          "gridTemplateColumns": "1fr 1fr",
          "marginBottom": "1em",
        }
      }
    >
      <span
        style={
          Object {
            "marginBottom": "1em",
          }
        }
      >
        Fornavn:
      </span>
      <span>
        Planet
      </span>
      <span
        style={
          Object {
            "marginBottom": "1em",
          }
        }
      >
        Mellomnavn:
      </span>
      <span>
        Planetus
      </span>
      <span
        style={
          Object {
            "marginBottom": "1em",
          }
        }
      >
        Etternavn:
      </span>
      <span>
        Planetetus
      </span>
    </div>
    <div
      style={
        Object {
          "display": "grid",
          "gridTemplateColumns": "1fr 1fr",
          "marginBottom": "1em",
        }
      }
    >
      <span>
        Telefonnummer:
      </span>
      <span>
        12345678
      </span>
    </div>
    <div
      style={
        Object {
          "display": "grid",
          "gridTemplateColumns": "1fr 1fr",
          "marginBottom": "1em",
        }
      }
    >
      <span
        style={
          Object {
            "marginBottom": "1em",
          }
        }
      >
        Gateadresse:
      </span>
      <span>
        Melkeveien 5
      </span>
      <span>
        Bruksenhet:
      </span>
      <span>
        H105
      </span>
    </div>
    <div
      style={
        Object {
          "display": "grid",
          "gridTemplateColumns": "1fr 1fr",
          "marginBottom": "1em",
        }
      }
    >
      <span
        style={
          Object {
            "marginBottom": "1em",
          }
        }
      >
        Postnummer:
      </span>
      <span>
        0985
      </span>
      <span
        style={
          Object {
            "marginBottom": "1em",
          }
        }
      >
        Poststed:
      </span>
      <span>
        Melk
      </span>
      <span>
        Bokommune:
      </span>
      <span>
        Veien
      </span>
    </div>
    <div
      style={
        Object {
          "display": "grid",
          "gridTemplateColumns": "1fr 1fr",
          "marginBottom": "1em",
        }
      }
    >
      <span>
        Søkers statsborkerskap:
      </span>
      <span>
        Solen, Uranus
      </span>
    </div>
    <div
      style={
        Object {
          "display": "grid",
          "gridTemplateColumns": "1fr 1fr",
          "marginBottom": "1em",
        }
      }
    >
      <span
        style={
          Object {
            "marginBottom": "1em",
          }
        }
      >
        Er søker registrert som flyktning?
      </span>
      <span>
        Ja
      </span>
      <span>
        Bor søker fast i Norge?
      </span>
      <span>
        Ja
      </span>
    </div>
  </div>
  <div
    style={
      Object {
        "marginBottom": "2em",
      }
    }
  >
    <h2
      className="typo-undertittel"
      style={
        Object {
          "marginBottom": "1em",
        }
      }
    >
      Boforhold
    </h2>
    <div
      style={
        Object {
          "display": "grid",
          "gridTemplateColumns": "1fr 1fr",
          "marginBottom": "1em",
        }
      }
    >
      <span
        style={
          Object {
            "marginBottom": "1em",
          }
        }
      >
        Deler søker bolig med en annen voksen?
      </span>
      <span>
        Ja
      </span>
    </div>
    <div
      style={
        Object {
          "display": "grid",
          "gridTemplateColumns": "1fr 1fr",
          "marginBottom": "1em",
        }
      }
    >
      <span>
        Søker deler bolig med:
      </span>
      <span>
        Ektefelle/Samboer/Partner, Barn over 18
      </span>
    </div>
    <div
      style={
        Object {
          "display": "grid",
          "gridTemplateColumns": "1fr 1fr",
          "marginBottom": "1em",
        }
      }
    >
      <span>
        Opplysninger:
      </span>
      <span>
        <ol>
          <li
            style={
              Object {
                "marginBottom": "1em",
              }
            }
          >
            12345678901
             
            Pluto
          </li>
          <li
            style={
              Object {
                "marginBottom": "1em",
              }
            }
          >
            10987654321
             
            Jupiter
          </li>
        </ol>
      </span>
    </div>
  </div>
  <div
    style={
      Object {
        "marginBottom": "2em",
      }
    }
  >
    <h2
      className="typo-undertittel"
      style={
        Object {
          "marginBottom": "1em",
        }
      }
    >
      Utenlandsopphold
    </h2>
    <div
      style={
        Object {
          "display": "grid",
          "gridTemplateColumns": "1fr 1fr",
          "marginBottom": "1em",
        }
      }
    >
      <label>
        Utenlandsopphold:
      </label>
      <label>
         
        <ol>
          <li
            style={
              Object {
                "marginBottom": "1em",
              }
            }
          >
            01-02-2020
             -  
            02-02-2020
          </li>
        </ol>
      </label>
    </div>
    <div
      style={
        Object {
          "display": "grid",
          "gridTemplateColumns": "1fr 1fr",
          "marginBottom": "1em",
        }
      }
    >
      <span>
        Planlagt utenlandsopphold:
      </span>
      <span>
        <ol>
          <li
            style={
              Object {
                "marginBottom": "1em",
              }
            }
          >
            04-02-2020
             -  
            06-02-2020
          </li>
        </ol>
      </span>
    </div>
  </div>
  <div
    style={
      Object {
        "marginBottom": "2em",
      }
    }
  >
    <h2
      className="typo-undertittel"
      style={
        Object {
          "marginBottom": "1em",
        }
      }
    >
      Opplysninger om oppholdstillatelse
    </h2>
    <div
      style={
        Object {
          "display": "grid",
          "gridTemplateColumns": "1fr 1fr",
          "marginBottom": "1em",
        }
      }
    >
      <span>
        Har søker varig oppholdstillatelse:
      </span>
      <span>
        Nei
      </span>
    </div>
    <div
      style={
        Object {
          "display": "grid",
          "gridTemplateColumns": "1fr 1fr",
          "marginBottom": "1em",
        }
      }
    >
      <span>
        Utløpsdato: 
      </span>
      <span>
        21-02-2020
      </span>
    </div>
    <div
      style={
        Object {
          "display": "grid",
          "gridTemplateColumns": "1fr 1fr",
          "marginBottom": "1em",
        }
      }
    >
      <span>
        Har søker søkt om forlengelse?
      </span>
      <span>
        Ja
      </span>
    </div>
  </div>
  <div
    style={
      Object {
        "marginBottom": "2em",
      }
    }
  >
    <h2
      className="typo-undertittel"
      style={
        Object {
          "marginBottom": "1em",
        }
      }
    >
      Inntekt, pensjon, og formue
    </h2>
    <div
      style={
        Object {
          "display": "grid",
          "gridTemplateColumns": "1fr 1fr",
          "marginBottom": "1em",
        }
      }
    >
      <span
        style={
          Object {
            "marginBottom": "1em",
          }
        }
      >
        Har søker fremsatt krav om annen norsk eller utenlandsk ytelse/pensjon som ikke er avgjort?
      </span>
      <span>
        Ja
      </span>
      <span>
        Hva slags ytelse/pensjon
      </span>
      <span>
        penjon
      </span>
    </div>
    <div
      style={
        Object {
          "display": "grid",
          "gridTemplateColumns": "1fr 1fr",
          "marginBottom": "1em",
        }
      }
    >
      <span
        style={
          Object {
            "marginBottom": "1em",
          }
        }
      >
        Har du arbeidsinntekt/personinntekt?
      </span>
      <span>
        Ja
      </span>
      <span
        style={
          Object {
            "marginBottom": "1em",
          }
        }
      >
        Brutto beløp per år:
      </span>
      <span>
        500
      </span>
      <span
        style={
          Object {
            "marginBottom": "1em",
          }
        }
      >
        Har du pensjon?
      </span>
      <span>
        <ol>
          <li
            style={
              Object {
                "marginBottom": "1em",
              }
            }
          >
            PensjonsOrdningen
            , 
            250
          </li>
        </ol>
      </span>
      <span>
        Sum inntekt:
      </span>
      <span>
        750
      </span>
    </div>
    <div
      style={
        Object {
          "display": "grid",
          "gridTemplateColumns": "1fr 1fr",
          "marginBottom": "1em",
        }
      }
    >
      <span
        style={
          Object {
            "marginBottom": "1em",
          }
        }
      >
        Har du formue/eiendom
      </span>
      <span>
        Ja
      </span>
      <span
        style={
          Object {
            "marginBottom": "1em",
          }
        }
      >
        Har du finansformue?
      </span>
      <span>
        Nei
      </span>
      <span
        style={
          Object {
            "marginBottom": "1em",
          }
        }
      >
        <span>
          Totalbeløp formue:
        </span>
      </span>
      <span>
        100
      </span>
      <span
        style={
          Object {
            "marginBottom": "1em",
          }
        }
      >
        Har du annen formue/eiendom
      </span>
      <span>
        Ja
      </span>
      <span>
        <span>
          -
        </span>
      </span>
      <span>
        <ol>
          <li
            style={
              Object {
                "marginBottom": "1em",
              }
            }
          >
            Hytte
            ,
            50
          </li>
        </ol>
      </span>
    </div>
    <div
      style={
        Object {
          "display": "grid",
          "gridTemplateColumns": "1fr 1fr",
          "marginBottom": "1em",
        }
      }
    >
      <span
        style={
          Object {
            "marginBottom": "1em",
          }
        }
      >
        Har søker Depositumskonto?
      </span>
      <span>
        Nei
      </span>
      <span />
      <span />
    </div>
    <div
      style={
        Object {
          "display": "grid",
          "gridTemplateColumns": "1fr 1fr",
          "marginBottom": "1em",
        }
      }
    >
      <span>
        Mottar du eller ektefellen/samboer, eller har du eller han/hun i løpet av de siste tre månedene mottatt sosialstønad til livsopphold?
      </span>
      <span>
        Nei
      </span>
    </div>
  </div>
  <div
    style={
      Object {
        "marginBottom": "2em",
      }
    }
  >
    <h2
      className="typo-undertittel"
      style={
        Object {
          "marginBottom": "1em",
        }
      }
    >
      For NAV
    </h2>
    <div
      style={
        Object {
          "display": "grid",
          "gridTemplateColumns": "1fr 1fr",
          "marginBottom": "1em",
        }
      }
    >
      <span>
        Hvilket målform ønsker du svaret i?
      </span>
      <span>
        Bokmål
      </span>
    </div>
    <div
      style={
        Object {
          "display": "grid",
          "gridTemplateColumns": "1fr 1fr",
          "marginBottom": "1em",
        }
      }
    >
      <span
        style={
          Object {
            "marginBottom": "1em",
          }
        }
      >
        Har søker møtt personlig?
      </span>
      <span>
        Ja
      </span>
      <span
        style={
          Object {
            "marginBottom": "1em",
          }
        }
      >
        Har fullmektig møtt?
      </span>
      <span>
        Nei
      </span>
      <span
        style={
          Object {
            "marginBottom": "1em",
          }
        }
      >
        Er originalt(e) pass sjekket for stempel?
      </span>
      <span>
        Ja
      </span>
    </div>
    <div
      style={
        Object {
          "display": "grid",
          "gridTemplateColumns": "1fr 1fr",
          "marginBottom": "1em",
        }
      }
    >
      <span
        style={
          Object {
            "marginBottom": "1em",
          }
        }
      >
        <span>
          Merknader:
        </span>
      </span>
      <span>
        <span>
          Alt er fint
        </span>
      </span>
    </div>
  </div>
</div>
`;
